### ISSUE TITLE
Feature Request: Total time to run all tests

### ISSUE THREAD

**Issue by RonanCodes** [Evilweed/protractor-beautiful-reporter#120]
It would be good to have the total time to run all tests shown on the report/
Perhaps we could extend this to show other totals.

**Comment by miller45** [comment]
You have this in the documents title if you include the reporter likes this:
```
onPrepare: function () {
        const startTime = new Date();
        // Add a screenshot reporter:
        jasmine.getEnv().addReporter(new HtmlReporter({
            baseDirectory: 'reports-tmp',
            docTitle: function () {
                const finishTime = new Date();
                const diffSeconds = Math.floor((finishTime - startTime) / 1000);

                return `Test Results (Total Duration ${diffSeconds}s)`;
            }
        }).getJasmine2Reporter());
    },
```

**Comment by RonanC** [comment]
This works but it could be a nice boolean option to add into the configuration options.

**Comment by miller45** [comment]
Yes of course: I am planning to add these small features with add-ons I am developing right now.... 
Reason: I do not want to ruin a nightly test runs of thousands of teams over world with adding bugs because of small features...so expect some lead time with this feature and the multiline feature

**Comment by vickyru** [comment]
@miller45 I also tried the same fix but problem is here if i have more than one spec (i.e. describe block) then for only first test case (i.e it block) total duration are getting calculated.

Below is the piece of code i tried

 onPrepare: function () {
      var timeStampFormatted = dateUtil.currentTimeStamp();
      const startTime = new Date();
      jasmine.getEnv().addReporter(new HtmlReporter({
              baseDirectory: './results',
              docName: 'results-' + timeStampFormatted + '.html',
              docTitle: function () {
                const finishTime = new Date();
                var diffSeconds = Math.floor((finishTime - startTime) / 1000);
                diffSeconds = Number(diffSeconds);
                var h = Math.floor(diffSeconds / 3600);
                var m = Math.floor(diffSeconds % 3600 / 60);
                var s = Math.floor(diffSeconds % 3600 % 60);

                var hDisplay = h > 0 ? h + (h == 1 ? "H" : "H") : "";
                var mDisplay = m > 0 ? m + (m == 1 ? "M" : "M") : "";
                var sDisplay = s > 0 ? s + (s == 1 ? "S" : "S") : "";

                return `Test Report (Duration ${hDisplay}:${mDisplay}:${sDisplay})`;
              },
              pathBuilder: function pathBuilder(spec, descriptions, results, capabilities) {
                  var browserName = capabilities.get('browserName');
                  var browserVersion = capabilities.get('version');
                  return path.join(
                      browserName +
                      '-' +
                      timeStampFormatted,
                      makeuuid());
              }
          }).getJasmine2Reporter(),
      );

  }
![duration](https://user-images.githubusercontent.com/5797680/50331848-2dfd9380-0526-11e9-84bf-a77ed64d2ef2.png)

Please do help me.

**Comment by vickyru** [comment]
@RonanC @miller45 
Any update on this thread?

**Comment by miller45** [comment]
@vickyru I have a solution for this now...but I will have to wait for the "release manager" to release the next version of the reporter.
